### ISSUE TITLE
Implement From trait for String

### ISSUE THREAD

**Issue by aleksator** [WiSaGaN/simple-error#10]
Consider the following function:
```
fn foo() -> SimpleResult<()> {
    File::open("non-existing-file").map_err(|err| format!("Error happened: {}", err))?;
    Ok(())
}
```
This code results in `the trait 'std::convert::From<&std::string::String>' is not implemented for 'simple_error::SimpleError'`

To work around that, explicit call to the constructor has to made:
```
fn foo() -> SimpleResult<()> {
    File::open("non-existing-file").map_err(|err| SimpleError::new(format!("Error happened: {}", err)))?;
    Ok(())
}
```

This is a bit verbose and could be alleviated by implementing a From trait for a String.

I know there is an existing implementation of From for &str, however, I could not figure out how to make it coerce in the example above.

**Comment by WiSaGaN** [comment]
```rust
fn foo() -> SimpleResult<()> {
    try_with!(File::open("non-existing-file"), "Error happened");
    Ok(())
}
```
Does this work for you?

**Comment by aleksator** [comment]
It can work in simple cases like this indeed.

However, when it's in a function chain, you naturally reach for .map_err()? in the middle.
Moreover, try_with! imposes the formatting of ", " between a message and an error, which is not always desirable.

**Comment by WiSaGaN** [comment]
Does this work for you?
```rust
fn foo() -> SimpleResult<()> {
    File::open("non-existing-file").map_err(|err| bail!("Error happened: {}", err))?;
    Ok(())
}
```

**Comment by aleksator** [comment]
Hi @WiSaGaN! Unfortunately, this code does not compile. The error is:

```
   |
20 | fn foo() -> SimpleResult<()> {
   |             ---------------- expected `simple_error::SimpleError` because of this
21 |     File::open("non-existing-file").map_err(|err| bail!("Error happened: {}", err))?;
   |                                                                                    ^ the trait `std::convert::From<std::result::Result<_, _>>` is not implemented for `simple_error::SimpleError`
   |
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
   = help: the following implementations were found:
             <simple_error::SimpleError as std::convert::From<&'a str>>
   = note: required by `std::convert::From::from`

error: aborting due to previous error
```

**Comment by WiSaGaN** [comment]
Sorry, I have released a new version `simple-error 0.2.2` (fixed in https://github.com/WiSaGaN/simple-error/commit/82217b84fbd9a0e7c960108a2d77a626651035c6), which supports a new macro `simple_error` for this usage:
```rust
fn foo() -> SimpleResult<()> {
    std::fs::File::open("non-existing-file").map_err(|err| simple_error!("Error happened: {}", err))?;
    Ok(())
}
```
